kwok/kwok_script_runner_test.go

```diff
@@ -0,0 +1,118 @@
+package kwok
+
+import (
+	_ "embed"
+	"os"
+	"testing"
+
+	"github.com/redhat-appstudio/segment-bridge.git/containerfixture"
+	"github.com/stretchr/testify/assert"
+)
+
+//go:embed kwok_container_template.tmpl
+var kwokServiceManifest string
+
+func configureNamespacesforContexts() {
+	useContext("kwok-host")
+	runOcCommand("apply", "-f", "./test_host_namespace_map.yml")
+	useContext("kwok-m01")
+	runOcCommand("apply", "-f", "test_member_namespace_map.yml")
+	useContext("kwok-rh01")
+	runOcCommand("apply", "-f", "test_rh_namespace_map.yml")
+}
+
+func TestGetUIDMap(t *testing.T) {
+	containerfixture.WithServiceContainer(t, kwokServiceManifest, func(deployment containerfixture.FixtureInfo) {
+		setKubeconfig("")
+
+		scriptPath := "../scripts/get-uid-map.sh"
+		executor := &ShellScriptExecutor{}
+		output, err := ExecuteScript(executor, scriptPath)
+		assert.NoError(t, err, "Failed to run getuidmap script")
+		outputMap, err := ParseScript(output)
+		assert.NoError(t, err, "Failed to run getuidmap script")
+
+		testCases := []struct {
+			name   string
+			input  map[string]interface{}
+			expect bool
+		}{
+			{"Usersignup validation", outputMap, true}, // using 'm' returned from setUpClusterConfiguration
+			{"Correct validation", map[string]interface{}{"petta": 12345678, "blee1": 87654321}, true},
+			{"empty key", map[string]interface{}{"": 12345678}, false},
+			{"no value key", map[string]interface{}{"<no value>": 87654321}, false},
+			{"empty map", map[string]interface{}{}, false},
+		}
+
+		for _, tc := range testCases {
+			t.Run(tc.name, func(t *testing.T) {
+				assert.Equal(t, tc.expect, validateUIDMap(tc.input), "Test case: %s", tc.name)
+			})
+		}
+	})
+}
+
+func TestGetWSMap(t *testing.T) {
```

Comment:
I do think get_usid_map and get_ws_map each deserve to have their own tests file that is names after them as opposed to naming the files after a mechanism used as a fixture.

Reply:
A. why not?
B. thy certainly do not belong in a`kwok` package. kwok is just one fixture mechanism out of several we have, and the scripts are actually major functional elements of segment-bridge.

Reply:
They are, but if anything- they probably belong more to the `scripts` package, where the scripts themselves exist,  than their own packages, where they don't have actual code running, but only the tests for the shell scripts.

Reply:
Noted, added 2 individual packages for the two scripts' tests.